tests/v2/validation/rbac/rbac_etcd_test.go

```diff
@@ -0,0 +1,132 @@
+package rbac
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning/rke2"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type ETCDRbacTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	standardUser       *management.User
+	standardUserClient *rancher.Client
+	session            *session.Session
+	cluster            *management.Cluster
+	adminProject       *management.Project
+	ns                 string
+	clusterName        string
+}
+
+func (rb *ETCDRbacTestSuite) TearDownSuite() {
+	rb.session.Cleanup()
+}
+
+func (rb *ETCDRbacTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	rb.session = testSession
+
+	rb.ns = defaultNamespace
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(rb.T(), err)
+
+	rb.client = client
+	rb.clusterName = client.RancherConfig.ClusterName
+	require.NotEmptyf(rb.T(), rb.clusterName, "Cluster name to install should be set")
+	clusterID, err := clusters.GetClusterIDByName(rb.client, rb.clusterName)
+	require.NoError(rb.T(), err, "Error getting cluster ID")
+	rb.cluster, err = rb.client.Management.Cluster.ByID(clusterID)
+	require.NoError(rb.T(), err)
+}
+
+func (rb *ETCDRbacTestSuite) ValidateEtcdSnapshotCluster(role string) {
+
+	/* Testcase1 Verify cluster members - Owner/member,  Project members - Owner/member are able to perform snapshot action on clusters
+	Cluster Owner/Restricted Admin - Allowed
+	Cluster Member/Project Owner/Project Member - Not Allowed */
+	log.Infof("Creating a snapshot of the cluster as %v", role)
+
+	err := rke2.CreateSnapshot(rb.standardUserClient, rb.clusterName, 1, rb.ns)
+
+	switch role {
+	case roleOwner, restrictedAdmin:
+		require.NoError(rb.T(), err)
+		// verify snapshot is created and is Active
+
+	case roleMember, roleProjectOwner, roleProjectMember:
+		require.Error(rb.T(), err)
+		strings.Contains(err.Error(), "is not updatable")
```

Comment:
lets add the full error like "Resource type [xxx] is not updatable", and assert the same error that we get from err.Error(). So, the code is more readable.

Reply:
done